format/Schema.fbs

```diff
@@ -183,10 +183,35 @@ table Timestamp {
 }
 
 enum IntervalUnit: short { YEAR_MONTH, DAY_TIME}
+// A "calendar" interval which models types that don't necessarily
+// have a precise duration without the context of a base timestamp (e.g.
+// days can differ in length during day light savings time transitions).
+// YEAR_MONTH - Indicates the number of elapsed whole months, stored as 
+//   4-byte integers. 
+// DAY_TIME - Indicates the number of elapsed days and milliseconds,
+//   stored as 2 contiguous 32-bit integers (8-bytes in total).  Support
+//   of this IntervalUnit is not required for full arrow compatibility.
```

Comment:
Got it. If it isn't being deprecated then it's reasonable to implement it in C++ and have integration tests